docs/data/data-grid/prompt/PromptWithExamples.tsx

```diff
@@ -0,0 +1,98 @@
+import * as React from 'react';
+import {
+  DataGridPremium,
+  Unstable_GridToolbarPromptControl as GridToolbarPromptControl,
+  GridToolbar,
+} from '@mui/x-data-grid-premium';
+import {
+  mockPromptResolver,
```

Comment:
I am not sure how much of a difference it would make, since you don't have to change/extend anything from it

Reply:
Since the users have to implement their own function similar to this, having the code readily available will give them an idea how it works. We could handle it as a follow-up too.